vm/package/deploy.rs

```diff
@@ -137,35 +138,46 @@ impl<N: Network> Package<N> {
     pub fn deploy<A: crate::circuit::Aleo<Network = N, BaseField = N::Field>>(
         &self,
         endpoint: Option<String>,
-    ) -> Result<Deployment<N>> {
-        // Retrieve the main program.
+    ) -> Result<()> {
         let program = self.program();
+        let mut process = Process::<N>::load()?;
+        self.add_imported_programs_to_process(&mut process, program.imports().keys())?;
+        if let Some(ref endpoint) = endpoint {
+            Self::deploy_program::<A>(&process, program, endpoint)?;
+        }
+        Ok(())
+    }
 
-        // Retrieve the program ID.
```

Comment:
Try to document each step like it's done previously. For example:

```
// Retrieve the program ID.
...

// Add the imported programs.

...

```

Reply:
I actually suggested him not to do that 😅 I think it makes sense when the comment helps clarifying the code below, but they are distracting when they just repeat what the code already does, e.g.

``` 
// Retrieve the program ID.
let program_id = program.id();
```

Reply:
In general, we like to err on the side of over documenting, so when api's change, we still know the exact intentions of the logic.

Reply:
Yeah I prefer to err on the side of over-documenting as well, but comments should complement or augment the information provided by the code, not reiterate it. E.g. establish intent, context and assumptions, especially at the interfaces (module, struct and function-level), and explaining/clarifying the complex parts of the implementation. From that perspective I'd say this module is under-documented even though it has many comments. Comments that just repeat the code are distracting to the reader and add maintenance burden without providing value.